lib/components/hikingmapbox.dart

```diff
@@ -167,30 +168,66 @@ class _MapWidgetState extends State<MapWidget> {
   }
 
   void updateRoute() async {
-    List<LatLng> remainingRoute = [];
-    List<LatLng> passedRoute = _passedRoute;
-
-    for (int i = 0; i < _route.length; i++) {
-      LatLng latLng = _route[i];
-      double distanceToCurrentLocation =
-          OsmData.getDistance(latLng, _currentDeviceLocation);
-      if (distanceToCurrentLocation > 0.0002) {
-        remainingRoute.add(latLng);
-      } else {
-        passedRoute.add(latLng);
+    int deleteAllRouteNodesWithAnIndexSmallerThan = 0;
+    for (int index = 0; index < _route.length; index++) {
+      if (shouldRouteNodeAtIndexBeConsideredForRemoval(index)) {
+        double distanceToCurrentLocation =
+            OsmData.getDistance(_route[index], _currentDeviceLocation);
+        if (distanceToCurrentLocation < 0.05) {
+          deleteAllRouteNodesWithAnIndexSmallerThan = index + 1;
```

Comment:
when changing the variable name with e.g. currentNodeIndex, this should be changed aswell:
currentNodeIndex = index;

Thus it is perfectly clear that when the distance to the user location is smaller than 0.05 the currentNodeIndex gets updated.